File: source/aoBlogs2/Views/BlogBodyView.vb

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Text.RegularExpressions
Imports Contensive.Addons.Blog.Controllers
Imports Contensive.Addons.Blog.Models
Imports Contensive.BaseClasses

Namespace Views
    '
    Public Class BlogBodyView
        '
        '========================================================================
        ''' <summary>
        ''' Inner Blog - the list of posts without sidebar. Blog object must be valid
        ''' </summary>
        ''' <returns></returns>
        Public Shared Function getBlogBody(cp As CPBaseClass, app As ApplicationController, legacyRequest As View.RequestModel, blogBodyRequest As BlogBodyRequestModel) As String
            Dim result As String = ""
            Try
                '
                ' Process Input
                Dim RetryCommentPost As Boolean = False
                Dim dstFormId As Integer = legacyRequest.FormID
                If legacyRequest.ButtonValue <> "" Then
                    '
                    ' Process the source form into form if there was a button - else keep formid
                    dstFormId = BlogBodyController.ProcessForm(cp, app, legacyRequest, RetryCommentPost)
                End If
                '
                ' -- Get Next Form
                result = GetForm(cp, app, legacyRequest, dstFormId, RetryCommentPost)
                '
                Return result
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
                Return String.Empty
            End Try
        End Function
        '
        '====================================================================================
        '
        Private Shared Function GetForm(cp As CPBaseClass, app As ApplicationController, request As View.RequestModel, dstFormID As Integer, RetryCommentPost As Boolean) As String
            Dim result As New StringBuilder()
            Try
                Select Case dstFormID
                    Case FormBlogPostDetails
                        result.Append(ArticleView.getArticleView(cp, app, request, RetryCommentPost))
                    Case FormBlogArchiveDateList
                        result.Append(ArchiveView.GetFormBlogArchiveDateList(cp, app, request))
                    Case FormBlogArchivedBlogs
                        result.Append(ArchiveView.GetFormBlogArchivedBlogs(cp, app, request))
                    Case FormBlogEntryEditor
                        result.Append(EditView.GetFormBlogEdit(cp, app, request))
                    Case FormBlogSearch
                        result.Append(SearchView.GetFormBlogSearch(cp, app, request))
                    Case Else
                        If (app.blogEntry IsNot Nothing) Then
                            '
                            ' Go to details page
                            '
                            result.Append(ArticleView.getArticleView(cp, app, request, RetryCommentPost))
                        Else
                            '
                            ' list all the entries
                            '
                            result.Append(ListView.GetListView(cp, app, request))
                        End If
                End Select
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result.ToString()
        End Function
        '
        '
    End Class
End Namespace

File: source/aoBlogs2/Views/BlogBodyView.cs

using Contensive.BaseClasses;
using Contensive.Blog.Models;
using System;
using System.Text;

namespace Contensive.Blog.Views {
    // 
    public class BlogBodyView {
        // 
        // ========================================================================
        /// <summary>
        /// Inner Blog - the list of posts without sidebar. Blog object must be valid
        /// </summary>
        /// <returns></returns>
        public static string getBlogBody(CPBaseClass cp, ApplicationEnvironmentModel app, Models.View.RequestModel legacyRequest, BlogBodyRequestModel blogBodyRequest) {
            string result;
            try {
                // 
                // Process Input
                bool RetryCommentPost = false;
                int dstFormId = legacyRequest.FormID;
                if (!string.IsNullOrEmpty(legacyRequest.ButtonValue)) {
                    // 
                    // Process the source form into form if there was a button - else keep formid
                    dstFormId = Controllers.BlogBodyController.ProcessForm(cp, app, legacyRequest, ref RetryCommentPost);
                }
                // 
                // -- Get Next Form
                result = GetForm(cp, app, legacyRequest, dstFormId, RetryCommentPost);
                // 
                return result;
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                return string.Empty;
            }
        }
        // 
        // ====================================================================================
        // 
        private static string GetForm(CPBaseClass cp, ApplicationEnvironmentModel app, Models.View.RequestModel request, int dstFormID, bool RetryCommentPost) {
            var result = new StringBuilder();
            try {
                switch (dstFormID) {
                    case constants.FormBlogPostDetails: {
                            result.Append(ArticleView.getArticleView(cp, app, RetryCommentPost));
                            break;
                        }
                    case constants.FormBlogArchiveDateList: {
                            result.Append(ArchiveView.getFormBlogArchiveDateList(cp, app, request));
                            break;
                        }
                    case constants.FormBlogArchivedBlogs: {
                            result.Append(ArchiveView.getFormBlogArchivedBlogs(cp, app, request));
                            break;
                        }
                    case constants.FormBlogEntryEditor: {
                            result.Append(EditView.getFormBlogEdit(cp, app, request));
                            break;
                        }
                    case constants.FormBlogSearch: {
                            result.Append(SearchView.GetFormBlogSearch(cp, app, request));
                            break;
                        }

                    default: {
                            if (app.blogPost is not null) {
                                // 
                                // Go to details page
                                // 
                                result.Append(ArticleView.getArticleView(cp, app, RetryCommentPost));
                            }
                            else {
                                // 
                                // list all the entries
                                // 
                                result.Append(ListView.getListView(cp, app, request));
                            }

                            break;
                        }
                }
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return result.ToString();
        }
        // 
        // 
    }
}